modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -366,6 +391,47 @@ private void onElectedAsLeader(long term) {
                         LOG.warn("Error when executing onLeaderElected callback", e);
                     }
                 });
+
+        raftServiceAfterJoin().thenCompose(this::pushAuthenticationConfigToCluster);
+
+    }
+
+    private CompletableFuture<Void> pushAuthenticationConfigToCluster(CmgRaftService service) {
+        return service.readClusterState()
+                .thenCompose(state -> {
+                    if (state == null) {
+                        LOG.info("No CMG state found in the Raft storage");
+                        return completedFuture(null);
+                    } else if (state.restAuthToApply() == null) {
+                        // auth config has already been successfully pushed to the distributed configuration
+                        LOG.info("No REST auth configuration found in the Raft storage");
+                        return completedFuture(null);
+                    } else {
+                        LOG.info("REST auth configuration found in the Raft storage, going to apply it");
+                        RestAuthentication restAuthToApply = state.restAuthToApply();
+                        return distributedConfigurationUpdater.updateRestAuthConfiguration(toRestAuthenticationConfig(restAuthToApply))
```

Comment:
I can see that `toRestAuthenticationConfig` is only needed to pass `RestAuthenticationConfig` to `DistributedConfigurationUpdater`. Maybe we should pass the `RestAuthentication` directly to the `DistributedConfigurationUpdater`? (This way we can also get rid of both `RestAuthenticationConfig` and `RestAuthConverter` classes)

Reply:
RestAuthenticationConfig is used in java API. RestAuthentication is used for messaging. So we need both. 
About DistributedConfigurationUpdater, I would like to avoid using messaging classes there and use different classes for different application layers.